internal/nitrite.go

```diff
@@ -197,55 +173,15 @@ func Verify(
 	}
 
 	doc := Document{}
-
 	err = cbor.Unmarshal(cose.Payload, &doc)
 	if nil != err {
-		return nil, ErrBadAttestationDocument
-	}
-
-	if "" == doc.ModuleID ||
-		"" == doc.Digest ||
-		0 == doc.Timestamp ||
-		nil == doc.PCRs ||
-		nil == doc.Certificate {
-		return nil, ErrMandatoryFieldsMissing
-	}
-
-	cert, err := x509.ParseCertificate(doc.Certificate)
-	if nil != err {
-		return nil, err
-	}
-
-	// TODO: remove the support for self-signed attestations as part of
-	//  https://blocky.atlassian.net/browse/BKY-5620 (remove !cert.IsCA path)
-	if !cert.IsCA && nil == doc.CABundle {
-		return nil, ErrMandatoryFieldsMissing
+		return nil, fmt.Errorf("unmarshaling document: %w", err)
 	}
 
-	if "SHA384" != doc.Digest {
-		return nil, ErrBadDigest
-	}
-
-	if doc.Timestamp < 1 {
-		return nil, ErrBadTimestamp
-	}
-
-	if len(doc.PCRs) < 1 || len(doc.PCRs) > 32 {
-		return nil, ErrBadPCRs
-	}
-
-	for key, value := range doc.PCRs {
-		if key > 31 {
-			return nil, ErrBadPCRIndex
-		}
+	cert, certificates, err := doc.Verify(certProvider, verificationTime)
```

Comment:
```suggestion
	_, certificates, err := doc.Verify(certProvider, verificationTime)
```

Reply:
Then below I changed
```go
	signatureOk := checkECDSASignature(
		cert.PublicKey.(*ecdsa.PublicKey),
		sigStruct,
		cose.Signature,
	)
```
to
```go
	signatureOk := checkECDSASignature(
		certificates[0].PublicKey.(*ecdsa.PublicKey),
		sigStruct,
		cose.Signature,
	)
```

Reply:
Obviously this would take updating the return type of `Document.Verify` and some error checking to make sure that `certificates` are not empty, but I ran this on my machine with these changes and all the tests pass, so I don't think you need to return `cert` separately.

Reply:
Updated signatures and added length check to cert chain before using